src/lpython/parser/parser.yy

```diff
@@ -488,13 +497,13 @@ if_statement
     ;
 
 for_statement
-    : KW_FOR tuple_item KW_IN expr ":" sep statements {
+    : KW_FOR tuple_for_statement_item KW_IN expr ":" sep statements {
         $$ = FOR_01($2, $4, $7, @$); }
-    | KW_FOR tuple_item KW_IN expr ":" sep statements KW_ELSE ":"
+    | KW_FOR tuple_for_statement_item KW_IN expr ":" sep statements KW_ELSE ":"
         sep statements { $$ = FOR_02($2, $4, $7, $11, @$); }
-    | KW_FOR tuple_item KW_IN expr ":" TK_TYPE_COMMENT TK_NEWLINE statements {
+    | KW_FOR tuple_for_statement_item KW_IN expr ":" TK_TYPE_COMMENT TK_NEWLINE statements {
         $$ = FOR_03($2, $4, $6, $8, @$); }
-    | KW_FOR tuple_item KW_IN expr ":" TK_TYPE_COMMENT TK_NEWLINE statements
+    | KW_FOR tuple_for_statement_item KW_IN expr ":" TK_TYPE_COMMENT TK_NEWLINE statements
```

Comment:
What do reviewers feel about this solution? If not, it's the precedence that needs to be separated for an expression and for a for/async for statements. Hence this separation, if there is a cleaner way to do this, please let me know.

cc: @Thirumalai-Shaktivel @akshanshbhatt

Reply:
Excellent work @krshrimali! Thanks for making the changes!
I think this is fine.

Reply:
Awesome, thanks for pointing out the mistake earlier. I appreciate it.

Reply:
I agree with @Thirumalai-Shaktivel. This looks neat. Excellent!